src/index.ts

```diff
@@ -69,6 +79,15 @@ async function main() {
         });
         console.log(`Cancel run ${id} responded with status ${res.status}`);
       }
+      if (all_but_latest && new Date(current_run.created_at) < cancel_before) {
```

Comment:
What's the purpose of this `if` statement?

Reply:
I can't have a run cancel itself before it has had a chance to cancel others. However, I think https://github.com/styfle/cancel-workflow-action/pull/35/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R69 is a bit sloppy and might break this if you're not using `all_but_latest`.

Reply:
Ah, no, if you're not `all_but_latest` you will never cancel `current_run`.

Reply:
Why would this run cancel itself? Shouldn't it always allow itself to run? Otherwise you would have nothing running, everything would cancel everything.

Reply:
It should cancel itself if it's not the most recent run